src/coreclr/jit/emitriscv64.cpp

```diff
@@ -3599,22 +3614,42 @@ void emitter::emitDispInsName(
                 case 0x0: // ADDIW
                     printf("addiw          %s, %s, %d\n", rd, rs1, imm12);
                     return;
-                case 0x1:                                                         // SLLIW
-                    printf("slliw          %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6 BITS for SHAMT in RISCV64
+                case 0x1: // SLLIW
+                {
+                    unsigned funct7 = (imm12 >> 5) & 0x7f;
+                    // SLLIW's instruction code's upper 7 bits have to be equal to zero
+                    if (funct7 == 0)
+                    {
+                        printf("slliw          %s, %s, %d\n", rd, rs1, imm12 & 0x1f); // 5 BITS for SHAMT in RISCV64
+                    }
+                    else
+                    {
+                        emitDispIllegalInstruction(code);
+                    }
+                }
                     return;
                 case 0x5: // SRLIW & SRAIW
-                    if (((code >> 30) & 0x1) == 0)
+                {
+                    static constexpr unsigned kLogicalShiftFunct7    = 0x00;
+                    static constexpr unsigned kArithmeticShiftFunct7 = 0x20;
```

Comment:
```suggestion
                    static constexpr unsigned kArithmeticShiftFunct7 = 0b0100000;
```
since -std=gnu++11 allows binary literals, we might as well use them